src/include/daos_task.h

```diff
@@ -37,6 +37,7 @@ extern "C" {
 #include <daos_errno.h>
 #include <daos_prop.h>
 #include <daos/tse.h>
+#include <daos/dtx.h>
```

Comment:
this is a public header, and the data structures are public.
I don't think we should add DTX ID which is private in those?

Reply:
Basically, I agree with this. I put it here just because I want to make the @dti to become one of dc_obj_update() input parameter. Currently, all parameters are attached to the task private buffer when create the task.

On the other hand, xuezhao is debug this issue. It seems that the shard_auxi_args is overwritten by accident. If he can find out more clew, this patch may be unnecessary. Personally, I do not want to use this hack solution.